File: elixir/resistor-color-duo/lib/resistor_color_duo.ex
defmodule ResistorColorDuo do
  @doc """
  Calculate a resistance value from two colors
  """
  @spec value(colors :: [atom]) :: integer
  def value(colors) do
    [first | [second | _]] = colors
    10 * color_code(first) + color_code(second)
  end

  @spec color_code(color :: atom) :: integer()
  def color_code(color) do
    case color do
      :black  -> 0
      :brown  -> 1
      :red    -> 2
      :orange -> 3
      :yellow -> 4
      :green  -> 5
      :blue   -> 6
      :violet -> 7
      :grey   -> 8
      :white  -> 9
      _       -> raise "invalid color"
    end
  end
end

File: csharp/resistor-color-duo/ResistorColorDuo.cs
using System;

public static class ResistorColorDuo
{
    private enum ColorCodes
    {
        black,
        brown,
        red,
        orange,
        yellow,
        green,
        blue,
        violet,
        grey,
        white,
    }
    public static int ColorCode(string color)
    {
        // can even make it case insensitive with true as second argument.
        return (int)Enum.Parse<ColorCodes>(color, true);
    }
    public static int Value(string[] colors)
    {
        return 10 * ColorCode(colors[0]) + ColorCode(colors[1]);
    }
}
